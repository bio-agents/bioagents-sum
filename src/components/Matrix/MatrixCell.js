import React from 'react'
import { Image } from 'react-bootstrap'

const MatrixCell = ({ image, text }) => {
  return (
    <div className={`square ${image || text ? 'dark-grey-background' : 'white-background'}`}>
      <div className='square-content'>
        <div className='table'>
          <div className='table-cell text-big'>
            {image && <Image src={image} rounded className='responsive-img' />}
            {text && <span>{text}</span>}
          </div>
        </div>
      </div>
    </div>
  )
}

export default MatrixCell
