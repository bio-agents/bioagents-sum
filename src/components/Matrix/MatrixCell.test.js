import React from 'react'
import renderer from 'react-test-renderer'
import MatrixCell from './MatrixCell'

it('Renders MatrixCell correctly', () => {
  const component = renderer.create(
    <MatrixCell />
  )

  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

it('Renders MatrixCell with name correctly', () => {
  const component = renderer.create(
    <MatrixCell name='A cell' />
  )

  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
