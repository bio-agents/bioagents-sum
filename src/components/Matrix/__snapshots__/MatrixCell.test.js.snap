exports[`test Renders MatrixCell correctly 1`] = `
<div
  className="square white-background">
  <div
    className="square-content">
    <div
      className="table">
      <div
        className="table-cell text-big" />
    </div>
  </div>
</div>
`;

exports[`test Renders MatrixCell with name correctly 1`] = `
<div
  className="square white-background">
  <div
    className="square-content">
    <div
      className="table">
      <div
        className="table-cell text-big" />
    </div>
  </div>
</div>
`;
